packages/connect-popup/src/index.tsx

```diff
@@ -454,7 +454,10 @@ const handshake = (handshake: PopupHandshake, origin: string) => {
 
     // when this message comes from iframe, settings is already validated.
     // when there is no iframe, we must validate it here
-    const trustedSettings = parseConnectSettings(payload.settings, origin);
+    const trustedSettings = parseConnectSettings(
+        payload.settings,
+        origin || payload.settings.origin || '',
```

Comment:
and it happens only when injecting trezor-connect-src right?

Reply:
It happens only when host and connect source are different. For example here https://staging-connect.trezor.io/9/#/method/getPublicKey it does not use `trezor-connect-src` but it happens becuse it uses https://suite.corp.sldev.cz/connect/release/connect/9.1.6/popup.html as connect source.

Reply:
After digging deep I think the root cause of it might be related to the use of BroadcastChannel that has same API as  postMessage but it looks like some arguments are not included and one of those is origin, that takes it's default value that is empty string - https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent/MessageEvent#origin

https://github.com/trezor/trezor-suite/blob/develop/packages/connect-iframe/src/index.ts#L235

https://stackoverflow.com/questions/63143886/is-there-a-way-to-reply-to-only-the-sender-after-receiving-a-broadcastchannel-m
![image](https://github.com/trezor/trezor-suite/assets/5362163/d6667170-985f-4bcf-878c-49111e643f08)

https://html.spec.whatwg.org/multipage/web-messaging.html#broadcasting-to-other-browsing-contexts

I am still investigating.  :mag_right:

Reply:
But I am not sure, because according to mdn message event from a BroadcastChannel CAN have origin https://developer.mozilla.org/en-US/docs/Web/API/BroadcastChannel/message_event

So maybe we have to set it because it is optional.